astroid/modutils.py

```diff
@@ -574,21 +574,12 @@ def is_relative(modname, from_file):
         from_file = os.path.dirname(from_file)
     if from_file in sys.path:
         return False
-    name = os.path.basename(from_file)
-    file_path = os.path.dirname(from_file)
-    parent_spec = importlib.util.find_spec(name, from_file)
-    while parent_spec is None and len(file_path) > 0:
-        name = os.path.basename(file_path) + "." + name
-        file_path = os.path.dirname(file_path)
-        parent_spec = importlib.util.find_spec(name, from_file)
-
-    if parent_spec is None:
-        return False
-
-    submodule_spec = importlib.util.find_spec(
-        name + "." + modname.split(".")[0], parent_spec.submodule_search_locations
+    modspec = importlib.machinery.PathFinder().find_spec(
+        modname.split(".")[0], [from_file]
```

Comment:
From what i can read in the `PathFinder` doc, it seems to do the job. However i would like to have @pkolbus opinion on this. Maybe he could tell us if did not use the `PathFInder` for a specific reason. 
Instead of `split` you can use `partition`. Performance is a bit better as it doesn't split multiple times the string but just one. Moreover, as the `PathFinder` doc say, there is no need to instantiate this class as all its methods are `@classmethod`. 
We thus could directly write:
```
return importlib.machinery.PathFinder.find_spec(modname.partition(".")[0], [from_file])
```

Reply:
This part of #857 wasn’t my work actually, it came from @degustaf in #686. I didn’t run into any errors with it, so didn’t evaluate alternatives. @degustaf, can you provide insight here?

Reply:
> This part of #857 wasn’t my work actually, it came from @degustaf in #686. I didn’t run into any errors with it, so didn’t evaluate alternatives. @degustaf, can you provide insight here?

Thanks @pkolbus !